sky/provision/vsphere/common/cls_api_client.py

```diff
@@ -0,0 +1,102 @@
+"""CLS API Client
+"""
+
+from com.vmware.content.library.item.downloadsession_client import (
+    File as DownloadSessionFile)
+from com.vmware.content.library.item.updatesession_client import (
+    File as UpdateSessionFile)
+from com.vmware.content.library.item_client import Changes
+from com.vmware.content.library.item_client import DownloadSession
+from com.vmware.content.library.item_client import UpdateSession
+from com.vmware.content.library_client import Item
+from com.vmware.content.library_client import SubscribedItem
+from com.vmware.content.library_client import Subscriptions
+from com.vmware.content_client import Library
+from com.vmware.content_client import LocalLibrary
+from com.vmware.content_client import SubscribedLibrary
+from com.vmware.vcenter.iso_client import Image
+from com.vmware.vcenter.ovf_client import LibraryItem
+from com.vmware.vcenter.vm_template.library_items_client import CheckOuts
+from com.vmware.vcenter.vm_template.library_items_client import Versions
+from com.vmware.vcenter.vm_template_client import LibraryItems
+from com.vmware.vcenter_client import VM
```

Comment:
For these imports, we should move them to the `sky/adaptors`, i.e., it should not be directly imported, but only be imported when being used. Reason: for a user without `vmware` dependencies installed, they will experience an `ImportError` as python will import these modules when `__init__` exists.

Reply:
Should we convert all files in sky/provision/vsphere/common into methods inside adaptors/vsphere.py in this case? Or simply move imports inside the functions?
I need clarification because files in sky/provision/vsphere/common were originally located in sky/skylet/providers/vsphere. We moved them because we were notified that sky/skylet/providers folder would be deprecated soon. Under sky/skylet/providers, imports like boto3 and googleapiclient were directly imported.

Reply:
Moving `sky/provision/vsphere/common` to `adaptors/vsphere.py` will not directly work as we still have to handle the imports with the functions. I am fine with importing the modules inside the functions, or we can create a adaptor function in `adaptors/vsphere.py` that returns the `com.vmware`.

```python
vmware = None

def import_package(func):

    @functools.wraps(func)
    def wrapper(*args, **kwargs):
        global vmware
        if vmware is None:
            try:
                import com.vmware as _vmware
            except ImportError:
                raise ImportError('Fail to import dependencies for Vsphere. '
                                  'Try pip install "skypilot[vsphere]"') from None
        return func(*args, **kwargs)

    return wrapper

@import_pacakge
def vmware():
    return _vmware
```

Reply:
Adaptors added